### PR TITLE
Docs part #2

### PR DESCRIPTION
**Pull request by bjoernricks**

### CODE REVIEW

File: docs/scripting.rst [hunk 1]
```diff
@@ -3,8 +3,131 @@
...
+  > gvm-cli socket --xml "<get_reports report_id=\"0f9ea6ca-abf5-4139-a772-cb68937cdfbb\"/>"
+  <get_reports_response status="200" status_text="OK"><report type="scan" id="0f9ea6ca-abf5-4139-a772-cb68937cdfbb" format_id="a994b278-1f62-11e1-96ac-406186ea4fc5" extension="xml" content_type="text/xml">
+  ...
+  </get_reports_response>
+
+See :command:`get_reports` command for all `details
+<https://docs.greenbone.net/API/OMP/omp.html#command_get_reports>`__.
+
+
+Additionally the report could be downloaded in a specific report format instead
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
Additionally, the report could be downloaded in a specific report format instead
```

File: docs/scripting.rst [hunk 2]
```diff
@@ -3,8 +3,131 @@
...
+of plain XML. Therefore all report formats can be listed with
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
of plain XML. All report formats can be listed with
```

File: docs/scripting.rst [hunk 3]
```diff
@@ -3,8 +3,131 @@
...
+
+A typical example for using the GMP protocol is the automatic scan of a new
+system. Below we assume that an Intrusion Detection System is in use that
+monitors the systems in the DMZ and immediately discovers new systems and
+unusual TCP ports not used up to now. If such an event is being discovered,
+the IDS should automatically initiate a scan of the new system. This can be
+done with the help of a script.
+
+Starting point is the IP address of the new suspected system. For this IP
+address a target needs to be created in the :term:`GSM`.
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
address, a target needs to be created in the :term:`GSM`.
```

File: docs/scripting.rst [hunk 4]
```diff
@@ -3,8 +3,131 @@
...
+If the IP address is saved in the environment variable :envvar:`IPADDRESS` by
+the IDS the respective target can be created with the following command:
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
the IDS, the respective target can be created with the following command:
```

File: docs/scripting.rst [hunk 5]
```diff
@@ -3,8 +3,131 @@
...
+.. code-block:: shell
+
+  > gvm-cli socket --xml "<create_task><name>Scan Suspect Host</name><target id=\"e5adc10c-71d0-49fe-aacf-a442ee31d387\"/><config id=\"daba56c8-73ec-11df-a475-002264764cea\"/></create_task>" 
+  <create_task_response status="201" status_text="OK, resource created" id="7249a07c-03e1-4197-99e4-a3a9ab5b7c3b"/>
+
+See :command:`create_task` command for all `details
+<https://docs.greenbone.net/API/OMP/omp.html#command_create_task>`__.
+
+
+Afterwards the task can be stated using the UUID return from the last response.
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
Afterwards the task can be started using the UUID return from the last response.
```

File: docs/tools.rst [hunk 6]
```diff
@@ -27,25 +27,180 @@ chosen. The other connection types require some setup and possible adjustments
...
 
 .. _gvm_cli:
 
 gvm-cli
 -------
 
+:program:`gvm-cli` is a low level tool which offers sending and receiving of
+commands and responses for the XML-based :term:`GMP (Greenbone Management
+Protocol) <GMP>` and :term:`OSP (Open Scanner Protocol) <OSP>` directly via the
+command line. It's intended for :ref:`simple scripting <xml_scripting>` via
```
Comment (wiegandm) [inline docs/tools.rst]:
```suggestion
command line. It is intended for :ref:`simple scripting <xml_scripting>` via
```

File: docs/tools.rst [hunk 7]
```diff
@@ -27,25 +27,180 @@ chosen. The other connection types require some setup and possible adjustments
...
+
+
 .. _gvm_script:
 
 gvm-script
 ----------
 
 .. versionadded:: 2.0
 
+The :program:`gvm-script` allows for running :ref:`gvm scripts <gvm_scripting>`
```
Comment (wiegandm) [inline docs/tools.rst]:
```suggestion
The :program:`gvm-script` allows running :ref:`gvm scripts <gvm_scripting>`
```

File: docs/scripting.rst [hunk 8]
```diff
@@ -3,8 +3,131 @@
...
+See :command:`get_report_formats` command for all `details
+<https://docs.greenbone.net/API/OMP/omp.html#command_get_report_formats>`__.
+
+E.g. to download the report in a PDF format the following command can be used:
+
+.. code-block:: shell
+
+  > gvm-cli socket --xml "<get_reports report_id=\"0f9ea6ca-abf5-4139-a772-cb68937cdfbb\" format_id=\"c402cc3e-b531-11e1-9163-406186ea4fc5\"/>"
+
+.. note:: Please be aware the PDF is returned as `base64 encoded
```
Comment (wiegandm) [inline docs/scripting.rst]:
```suggestion
.. note:: Please be aware that the PDF is returned as `base64 encoded
```

**Comment by bjoernricks** [comment]
Still not finished but IMHO the new content should be merged already.

Missing content:
* Connection type TLS
* Connection type SSH
* GVM Scripting

File: docs/connectiontypes.rst [hunk 9]
```diff
@@ -0,0 +1,65 @@
+.. _connection_types:
+
+Connection Types
+================
+
+Before being able to talk to a remote :term:`GMP` or :term:`OSP` server using
+one of the :ref:`provided command line clients <tools>` the the user
```
Comment (wiegandm) [inline docs/connectiontypes.rst]:
```suggestion
one of the :ref:`provided command line clients <tools>`, the user
```

File: docs/connectiontypes.rst [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+:file:`/usr/share/gvm/gsad/web/gvmd.sock`.
+
+:term:`OSPd based scanners <OSPd>` may be accessed via Unix Domain Sockets too.
+The location and name of these sockets is configurable and depends on the used
+OSPd scanner implementation.
+
+.. warning::
+
+  Accessing a Unix Domain Socket requires sufficient unix file permissions for
+  the user running the :ref:`command line interface tool <tools>`. Please don't
```
Comment (wiegandm) [inline docs/connectiontypes.rst]:
```suggestion
  the user running the :ref:`command line interface tool <tools>`. Please do not
```

File: docs/connectiontypes.rst [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  start a tool as **root** user via :command:`sudo` or :command:`su` only to
+  avoid not being able to access the socket path. Instead please adjust the
```
Comment (wiegandm) [inline docs/connectiontypes.rst]:
```suggestion
  be able to access the socket path. Instead, please adjust the
```

File: docs/glossary.rst [hunk 12]
```diff
@@ -41,6 +43,23 @@ Glossary
...
+  GVM10
+    `Version 10 <https://community.greenbone.net/t/gvm-10-beta/208>`_ of the
+    :term:`GVM` framework. Used in :term:`GOS 5 <GOS>`.
+
+  GSE
+    The `Greenbone Source Edition (GSE)
+    <https://community.greenbone.net/t/about-the-source-edition-gse-category/176>`_
+    covers the actual source codes of the Greenbone application stack
+    for vulnerability scanning and vulnerability management :term:`GVM`.
+    The source edition is adopted by external third parties. E.g. if your
```
Comment (wiegandm) [inline docs/glossary.rst]:
```suggestion
    The source edition is adopted by external third parties. E.g., if your
```

File: docs/glossary.rst [hunk 13]
```diff
@@ -41,6 +43,23 @@ Glossary
...
+    :term:`GVM` stack is provided by a Linux Distribution then it is build from
```
Comment (wiegandm) [inline docs/glossary.rst]:
```suggestion
    :term:`GVM` stack is provided by a Linux Distribution, it is build from
```

**Pull request merged**